pkg/middleware/auth.go

```diff
@@ -247,6 +248,25 @@ func SnapshotPublicModeOrSignedIn(cfg *setting.Cfg) web.Handler {
 			notAuthorized(c)
 			return
 		}
+
+		ac.Middleware(ac2)(ac.EvalPermission(dashboards.ActionSnapshotsCreate))
```

Comment:
I can see that snapshot endpoints use app platform k8s APIs if `FlagKubernetesSnapshots` is enabled. Do we need to carry out AC checks in that case?